akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/BehaviorTestKit.scala

```diff
@@ -60,6 +62,20 @@ object BehaviorTestKit {
 @ApiMayChange
 abstract class BehaviorTestKit[T] {
 
+  /**
+   * Constructs a message using the provided 'messageFactory' to inject a single-use "reply to"
+   * [[akka.actor.typed.ActorRef]], and sends the constructed message to the behavior, recording any [[Effect]]s.
+   *
+   * The returned [[TestInbox]] allows the message sent to the "reply to" `ActorRef` to be asserted on.
+   */
+  def ask[Res](messageFactory: JFunction[ActorRef[Res], T]): TestInbox[Res]
+
+  /**
+   * The same as [[ask]] but only for requests that result in a response of type [[akka.pattern.StatusReply]].
+   */
+  def askWithStatus[Res](messageFactory: JFunction[ActorRef[StatusReply[Res]], T]): TestInbox[StatusReply[Res]] =
+    ask[StatusReply[Res]](messageFactory)
```

Comment:
Alternatively, maybe a specialized `StatusReplyInbox` is called for?

Reply:
One idea could be to extract/return the successful response value, and throw the error if it was an error.